pkg/generator/generator.go

```diff
@@ -111,8 +133,99 @@ func NewGenerator(apiVersion, kind, projectName, repoPath string) *Generator {
 // │   |   ├── build
 // │   |   └── codegen
 // │   └── version
+//
+// Render generates the following project structure for Operator type Ansible
+// ├── <projectName>
+// │   ├── Dockerfile
+// │   ├── roles
+// │   │   └── <kind>
+// │   ├── watches.yaml
+// │   ├── deploy
+// │   │   ├── <kind>-CRD.yaml
+// │   │   ├── rbac.yaml
+// │   │   ├── operator.yaml
+// │   ├   └── cr.yaml
+
 func (g *Generator) Render() error {
-	if err := g.generateDirStructure(); err != nil {
+	switch g.operatorType {
+	case "ansible":
+		if err := g.renderAnsibleOperator(); err != nil {
+			return err
+		}
+	case "go":
+		if err := g.renderGoOperator(); err != nil {
+			return err
+		}
+	default:
+		return errors.New(fmt.Sprintf("unexpected operator type [%v]", g.operatorType))
+	}
+	return nil
+}
+
+func (g *Generator) renderAnsibleOperator() error {
+	if err := g.generateAnsibleDirStructure(); err != nil {
+		return err
+	}
+	if err := g.renderTmp(); err != nil {
+		return err
+	}
+	if err := g.renderWatches(); err != nil {
+		return err
+	}
+	if err := g.renderPlaybook(); err != nil {
+		return err
+	}
+	if err := g.renderDeploy(); err != nil {
+		return err
+	}
+	return g.renderRole()
+}
+
+func (g *Generator) renderRole() error {
+	fmt.Printf("Rendering Ansible Galaxy role [%v/%v/%v]...\n", g.projectName, rolesDir, g.kind)
+	agcmd := exec.Command(filepath.Join(g.projectName, galaxyInitCmd))
```

Comment:
Should we maybe remove this file?

Reply:
I chose to wrap the `ansible-galaxy init` command around a bash script to catch if `ansible` is not installed on the host. This seems to be the same strategy taken by the Docker build script so I kept it all in the `tmp/` folder.

Reply:
After talking with Shawn I think it makes sense to remove the `tmp/init` dir after creation.